src/DIRAC/FrameworkSystem/Client/ProxyGeneration.py

```diff
@@ -297,11 +297,14 @@ def generateProxy(params):
   # First try reading the key from the file
   retVal = testChain.loadKeyFromFile(params.keyLoc, password=params.userPasswd)  # XXX why so commented?
   if not retVal['OK']:
-    passwdPrompt = "Enter Certificate password:"
     if params.stdinPasswd:
       userPasswd = sys.stdin.readline().strip("\n")
     else:
-      userPasswd = getpass.getpass(passwdPrompt)
+      try:
+        userPasswd = prompt(u"Enter Certificate password: ", is_password=True)
+      except KeyboardInterrupt:
+        gLogger.error("Caught KeyboardInterrupt, exiting...")
+        sys.exit(1)
```

Comment:
You should return `S_ERROR` here, and the caller will call `sys.exit(1)`
```suggestion
        return S_ERROR("Caught KeyboardInterrupt, exiting...")
```